src/main/java/com/teragrep/cfe_16/EventManager.java

```diff
@@ -142,14 +143,14 @@ public ObjectNode convertData(
          * After the event is handled, it is assigned as a value to previousEvent
          * variable.
          */
-        HttpEventData eventData = null;
+        TimestampedHttpEventData eventData = null;
         Converter converter = new Converter();
-        List<SyslogMessage> syslogMessages = new ArrayList<SyslogMessage>();
+        List<SyslogMessage> syslogMessages = new ArrayList<>();
         while (parser.hasNext()) {
             previousEvent = eventData;
             String jsonObjectStr = parser.next().toString();
             eventData = verifyJsonData(jsonObjectStr, previousEvent);
-            eventData = assignMetaData(eventData, authToken, channel);
+            assignMetaData(eventData, authToken, channel);
```

Comment:
method is using setters to modify method call args, which is horrible. please remove the method and move at least the calling code here.